File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/recipes/domain/usecase/GetRecipeUseCaseTest.java
package com.tarbadev.witchcraft.recipes.domain.usecase;

import com.tarbadev.witchcraft.TestResources;
import com.tarbadev.witchcraft.recipes.domain.usecase.GetRecipeUseCase;
import com.tarbadev.witchcraft.recipes.domain.entity.Recipe;
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import static org.assertj.core.api.Java6Assertions.assertThat;
import static org.mockito.BDDMockito.given;

@RunWith(SpringRunner.class)
@SpringBootTest
@ActiveProfiles("test")
public class GetRecipeUseCaseTest {
    private GetRecipeUseCase subject;

    @Autowired private TestResources testResources;
    @Mock private RecipeRepository recipeRepository;

    @Before
    public void setUp() {
        subject = new GetRecipeUseCase(recipeRepository);
    }

    @Test
    public void execute() {
        Recipe recipe = testResources.getRecipe();

        Integer recipeId = 123;
        given(recipeRepository.findById(recipeId)).willReturn(recipe);

        assertThat(subject.execute(recipeId)).isEqualTo(recipe);
    }
}

File: WitchcraftApp/src/test/kotlin/com/tarbadev/witchcraft/recipes/domain/usecase/GetRecipeUseCaseTest.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import com.tarbadev.witchcraft.TestResources
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class GetRecipeUseCaseTest {
    private lateinit var subject: GetRecipeUseCase

    private val recipeRepository: RecipeRepository = mock()

    @BeforeEach
    fun setUp() {
        subject = GetRecipeUseCase(recipeRepository)
    }

    @Test
    fun execute() {
        val recipe = TestResources.Recipes.cookinCanuck

        val recipeId = 123
        whenever(recipeRepository.findById(recipeId)).thenReturn(recipe)

        assertEquals(recipe, subject.execute(recipeId))
    }
}